packages/slice-machine/components/CodeBlock/index.tsx

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { Flex } from "theme-ui";
+import hljs from "highlight.js";
+
+export type Language = "javascript" | "bash" | "xml" | "html" | "json";
+
+const DEFAULT_LANGUAGES: Array<Language> = [
+  "javascript",
+  "bash",
+  "xml",
+  "html",
+  "json",
+];
+
+const CodeBlock: React.FC<{
+  children: string;
+  lang?: Language;
+  style?: React.CSSProperties;
+}> = ({ children, lang, style }) => {
+  const text = lang
+    ? hljs.highlight(children, { language: lang }).value
+    : hljs.highlightAuto(children, DEFAULT_LANGUAGES).value;
+
+  return (
+    <Flex as="pre" style={style}>
+      <code
+        className="hljs"
+        style={{ overflowX: "auto", padding: "3px 5px" }}
+        dangerouslySetInnerHTML={{ __html: text }}
```

Comment:
isn't that dangerous ?
It's one of the things that isn't recommended for XSS.

Reply:
Yes it's dangerous, but most of the content is hard-coded in the hints a side from a few things like field id